kong/kong.lua

```diff
@@ -147,6 +148,28 @@ function Kong.init_worker()
   for _, plugin in ipairs(singletons.loaded_plugins) do
     plugin.handler:init_worker()
   end
+
+  local ev = require "resty.worker.events"
+  local handler = function(data, event, source, pid)
+    if source and source == constants.CACHE.CLUSTER then
+      singletons.events:publish(event, data)
+    end
+  end
+
+  ev.register(handler)
+
+  local ok, err = ev.configure {
+      shm = "process_events", -- defined by "lua_shared_dict"
+      timeout = 2,            -- life time of event data in shm
+      interval = 1,           -- poll interval (seconds)
+
+      wait_interval = 0.010,  -- wait before retry fetching event data
+      wait_max = 0.5,         -- max wait time before discarding event
+  }
+  if not ok then
+      ngx.log(ngx.ERR, "failed to start event system: ", err)
```

Comment:
this should be a hard error, not just logging. If this fails, Kong will not be stable, so it must error out and stop.